thinc/layers/with_ragged.py

```diff
@@ -18,20 +17,27 @@ def forward(
     model: Model[SeqT, SeqT], Xseq: SeqT, is_train: bool
 ) -> Tuple[SeqT, Callable]:
     layer: Model[Ragged, Ragged] = model.layers[0]
-    Y: Union[Padded, Ragged, List2d, RaggedData]
     if isinstance(Xseq, Ragged):
-        Y, backprop = layer(Xseq, is_train)
+        ragged_Y, backprop = layer(Xseq, is_train)
+        Y = cast(SeqT, ragged_Y)
     elif isinstance(Xseq, Padded):
-        Y, backprop = _padded_forward(layer, cast(Padded, Xseq), is_train)
+        padded_Y, backprop = _padded_forward(layer, cast(Padded, Xseq), is_train)
+        Y = cast(SeqT, padded_Y)
     elif _is_ragged_data(Xseq):
-        Y, backprop = _tuple_forward(layer, cast(RaggedData, Xseq), is_train)
+        ragged_data_Y, backprop = _tuple_forward(
+            layer, cast(RaggedData, Xseq), is_train
+        )
```

Comment:
Same comment as before - especially when code starts getting split across different lines, it's worth it making the var names shorter to try and get things on one line for readability